<?xml version="1.0"?>
<launch>

  
  #### JSK Quadcopter ###################################

  <group ns="sim" >
    <node pkg="jsk_quadcopter" type="jsk_quadcopter_node" name="quadcopter_node"  output="screen" > 

      <param name="simulationFlag" type="bool" value="true"/>

      <param name="rxLoopRate" type="double" value="110"/>
      <param name="txLoopRate" type="double" value="40"/>
      <param name="tfPubLoopRate" type="double" value="120"/>

      <!--                Simulation  -->
      <param name="simulationFlag" type="bool" value="true"/> <!-- receive  the sim rosbag file -->
    <param  name="navigator/useRocketStart" type="bool" value="true"/>

    ############### Quadctoper Basic State ######################
    <rosparam file="$(find jsk_quadcopter)/config/rook_p/EstimatorConfig.yaml" command="load" />

      <!--                Kalman Filter  -->
    <param name="estimator/kalmanFilterFlag" type="bool" value="true"/>
    <param name="kalman_filter/imuHz" type="double" value="105"/>
<!--    <rosparam  file="$(find jsk_quadcopter)/config/simulation/KduinoKalmanFilterConfig.yaml" command="load" /> -->
    <rosparam  file="$(find jsk_quadcopter)/config/simulation/OpticalFlowKalmanFilterConfig.yaml" command="load" />

    ############### Kalman Filter Debug/Simulation ###
    <param name="estimator/kalmanFilterDebug" type="bool" value="true"/>
    <param name="estimator/kalmanFilterAxis" type="int" value="1"/>
    <rosparam file="$(find jsk_quadcopter)/config/simulation/SimulationKalmanFilterConfig.yaml" command="load" />
    <rosparam file="$(find jsk_quadcopter)/config/simulation/RocketStartFreeFallConfig.yaml" command="load" />


    ############### Hokuyo and Mirror Module Param ###
    <rosparam file="$(find jsk_quadcopter)/config/rook2/MirrorLaserModuleConfig.yaml" command="load" />


      <!--                  IMU DATA      -->
      <remap from="/sim/imu/kduino/imu" to="/kduino/imu" />
      <remap from="/sim/mirror/scan" to="/scan" />
      <remap from="/sim/mirror/laser_boundary_offset" to="/sim/laser_boundary_offset" />
      <!--                  Optical Flow Data -->
      <remap from="/sim/opticalFlow/opt_flow" to="/px4flow/opt_flow" /> 

      <!--                  Slam Data -->
      <remap from="/sim/slam/slam_out_pose" to="/slam/slam_out_pose" />
    </node>
  </group>
</launch>



